File: src/main/java/net/hectus/neobb/turn/legacy_game/item/LTEnchantedGoldenApple.java
package net.hectus.neobb.turn.legacy_game.item;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.buff.ChancedBuff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.CounterFilter;
import net.hectus.neobb.turn.default_game.attributes.clazz.NeutralClazz;
import net.hectus.neobb.turn.default_game.attributes.function.BuffFunction;
import net.hectus.neobb.turn.default_game.attributes.function.CounterFunction;
import net.hectus.neobb.turn.default_game.item.ItemTurn;
import net.hectus.neobb.turn.default_game.mob.MobTurn;
import org.bukkit.Location;
import org.bukkit.inventory.ItemStack;

import java.util.List;

public class LTEnchantedGoldenApple extends ItemTurn implements CounterFunction, BuffFunction, NeutralClazz {
    public LTEnchantedGoldenApple(NeoPlayer player) { super(player); }
    public LTEnchantedGoldenApple(ItemStack data, Location location, NeoPlayer player) { super(data, location, player); }

    @Override
    public int cost() {
        return 6;
    }

    @Override
    public List<Buff> buffs() {
        return List.of(new ChancedBuff(new Buff.Luck(Buff.BuffTarget.YOU, 100), 50));
    }

    @Override
    public List<CounterFilter> counters() {
        return List.of(CounterFilter.of(t -> t instanceof MobTurn<?>, "mobs"));
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/legacy_game/item/LTEnchantedGoldenApple.kt
package net.hectus.neobb.turn.legacy_game.item

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.buff.ChancedBuff
import net.hectus.neobb.buff.Luck
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.turn.default_game.CounterFilter
import net.hectus.neobb.turn.default_game.attribute.clazz.NeutralClazz
import net.hectus.neobb.turn.default_game.attribute.function.BuffFunction
import net.hectus.neobb.turn.default_game.attribute.function.CounterFunction
import net.hectus.neobb.turn.default_game.item.ItemTurn
import net.hectus.neobb.turn.default_game.mob.MobTurn
import org.bukkit.inventory.ItemStack

class LTEnchantedGoldenApple(data: ItemStack?, cord: Cord?, player: NeoPlayer?) : ItemTurn(data, cord, player), CounterFunction, BuffFunction, NeutralClazz {
    override val cost: Int = 6

    override fun counters(): List<CounterFilter> {
        return listOf(CounterFilter.of("mobs") { it is MobTurn<*> })
    }

    override fun buffs(): List<Buff<*>> {
        return listOf(ChancedBuff(50.0, Luck(100)))
    }
}
